src/Pos/Util/Servant.hs

```diff
@@ -93,31 +98,29 @@ instance (HasServer (Verb mt st ct $ ApiModifiedRes mod a) ctx,
         handlerCatch = mapHandler $ modifyApiResult (Proxy @mod)
 
 -------------------------------------------------------------------------
--- Mapping API arguments
+-- Mapping API arguments: client types decoding
 -------------------------------------------------------------------------
 
--- | For many types with nice structure there exist a @CType@ - intermediate
--- form (more string-like in some sense) which has automatically derived
--- JSON instances and is used by daedalus-bridge.
-class FromCType c where
-   -- | For given @CType@ - relative original type
-   -- (e.g. @CAccountAddress@ - @AccountAddress@).
-   type family FromOriginType c :: *
-   type instance FromOriginType c = ToOriginType c
+-- | For many types with nice structure there exist a /client type/ -
+-- intermediate form (more string-like in some sense) which has automatically
+-- derived JSON instances and is used by daedalus-bridge.
+-- | This family, for given /client type/, gets relative original type
+-- (e.g. @CAccountAddress@ - @AccountAddress@).
+type family OriginType ctype :: *
 
+class FromCType c where
    -- | Way to decode from @CType@.
-   decodeCType :: c -> Either Text (FromOriginType c)
+   decodeCType :: c -> Either Text (OriginType c)
 
--- This is not used for now
 class ToCType c where
-   -- | Same as 'FromOriginType'. For some types they actually differ,
```

Comment:
Yes, previously the difference was in Maybe only, because there were mandatory and optional parameters in request, we wanted to specify special value in the latter case when parameter is not set (e.g. `emptyPasspharse`, if passphrase is not specified)